warehouse/tuf/tasks.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import redis
+
+from warehouse.tasks import task
+from warehouse.tuf import utils
+from warehouse.tuf.interfaces import IKeyService, IRepositoryService
+
+
+@task(bind=True, ignore_result=True, acks_late=True)
+def bump_role(task, request, role):
+    r = redis.StrictRedis.from_url(request.registry.settings["celery.scheduler_url"])
+
+    with utils.RepoLock(r):
```

Comment:
I don't think that you need another lock CM wrapper. The lib already exposes a context manager: https://github.com/andymccurdy/redis-py/blob/ce88fca/redis/lock.py#L151-L159.
```suggestion
    with r.lock("tuf-repo"):
```

Reply:
Looks good, thanks for finding that~